node-manager/src/node.rs

```diff
@@ -365,6 +367,114 @@ impl Node {
+        let invoice = Invoice::from_str(&invoice_str).map_err(Into::<MutinyError>::into)?;
+        let payment_hash = invoice.payment_hash();
+        let (payment_info, inbound) = self.get_payment_info_from_persisters(payment_hash)?;
+        let mut mutiny_invoice: MutinyInvoice = invoice.into();
+        mutiny_invoice.is_send = !inbound;
+        mutiny_invoice.paid = matches!(payment_info.status, HTLCStatus::Succeeded);
+        Ok(mutiny_invoice)
+    }
+
+    pub fn list_invoices(&self) -> Result<Vec<MutinyInvoice>, MutinyError> {
+        let mut inbound_invoices = self.list_payment_info_from_persisters(true);
+        let mut outbound_invoices = self.list_payment_info_from_persisters(false);
+        inbound_invoices.append(&mut outbound_invoices);
+        Ok(inbound_invoices)
+    }
+
+    fn list_payment_info_from_persisters(&self, inbound: bool) -> Vec<MutinyInvoice> {
+        self.persister
+            .list_payment_info(inbound)
+            .into_iter()
+            .filter_map(|(h, i)| match i.bolt11 {
+                Some(bolt11) => {
+                    // Construct an invoice from a bolt11, easy
+                    let invoice_res = Invoice::from_str(&bolt11).map_err(Into::<MutinyError>::into);
+                    match invoice_res {
+                        Ok(invoice) => {
+                            let mut mutiny_invoice: MutinyInvoice = invoice.into();
+                            mutiny_invoice.is_send = !inbound;
+                            mutiny_invoice.paid = matches!(i.status, HTLCStatus::Succeeded);
+                            Some(mutiny_invoice)
+                        }
+                        Err(_) => None,
+                    }
+                }
+                None => {
+                    // Constructing MutinyInvoice from no invoice, harder
+                    let paid = matches!(i.status, HTLCStatus::Succeeded);
+                    // TODO finish this properly
+                    let mutiny_invoice =
+                        MutinyInvoice::new(None, None, h, None, None, 0, paid, None, !inbound);
+                    Some(mutiny_invoice)
+                }
+            })
+            .collect()
+    }
+
+    fn get_payment_info_from_persisters(
+        &self,
+        payment_hash: &bitcoin_hashes::sha256::Hash,
+    ) -> Result<(PaymentInfo, bool), MutinyError> {
+        // try inbound first
+        if let Some(payment_info) = self
+            .persister
+            .read_payment_info(PaymentHash(payment_hash.into_inner()), true)
+        {
+            return Ok((payment_info, true));
+        }
+
+        // if no inbound check outbound
+        match self
+            .persister
+            .read_payment_info(PaymentHash(payment_hash.into_inner()), false)
+        {
+            Some(payment_info) => Ok((payment_info, false)),
+            None => Err(MutinyError::InvoiceInvalid),
+        }
+    }
+
+    // pay_invoice sends off the payment but does not wait for results
```

Comment:
This comment should have 3 slashes so it appears as docs for the function

Reply:
added